src/main/resources/generators/javascript.html

```diff
@@ -0,0 +1,36 @@
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+    
+    http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+<div id="${uniqueId}"></div>
+<script type="text/javascript">
+    if (window.XMLHttpRequest) {
+        var xhr = new XMLHttpRequest();
+        xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+        xhr.timeout = 10000;
```

Comment:
@SBPrakash - any particular reason to set the header and timeout?

Reply:
Header - for the purpose just to give an extra hand to developers incase if there is a need to check for header in apache or java..
Timer - page shouldn't be waiting for this request to complete more than 10sec..even 10sec is more, considering lesser bandwidth network may take some time to download resources.